config/src/test/scala/com/typesafe/config/impl/ConfigBeanFactoryTest.scala

```diff
@@ -178,6 +178,7 @@ class ConfigBeanFactoryTest extends TestUtils {
         assertNotNull(beanConfig)
         assertNotNull(beanConfig.getValueObject)
         assertNull(beanConfig.getValueObject.getOptionalValue)
+        assertNull(beanConfig.getValueObject.getNullableValue)
```

Comment:
Currently, we do kind of test the use case where a value is set to null versus a value that is simply missing or omitted. The `ObjectsConfig#optionalValue` field is being used to test the value being omitted, and the `ObjectsConfig#nullableValue` field for the value being set to null.